### ISSUE TITLE
Strange characters appearing near monster healthbars

### ISSUE THREAD

**Issue by lightrow** [bayaraa/d2gl#96]
D2GL 1.3.0
D2 1.09d + PlugY running on macos through crossover
seems to be hard to reproduce, happened twice but now no longer

https://github.com/bayaraa/d2gl/assets/11335104/e816bc68-b555-4a14-9ca6-fb0412ce8db0

**Comment by bayaraa** [comment]
hmm. strange one.
how about without plugy?

**Comment by rnd2k** [comment]
Hi! I saw similar bug, appears on vanilla versions 1.09 and 1.10 on the level barracks - jail act 1.
https://youtu.be/UTitYVoSjvw
I tried to fix it, but I don't know if it's correct.
```
void HDText::drawSubText(uint8_t fn)
{
	if (m_map_names) {
		m_map_names = false;
		return;
	}

	if (!isActive() || !d2::sub_text_ptr)
		return;

	const uintptr_t ptr = (uintptr_t)d2::sub_text_ptr;
	static const wchar_t* str;
	static uint32_t* color;
	static int *length, *x, *y;

	if (fn == 1) {
		if (isVerMax(V_110)) {
			str = *(const wchar_t**)(ptr + 0x20);
			color = (uint32_t*)(ptr + 0x78);
			length = (int*)(ptr + 0x1C);
			x = (int*)(ptr + 0x18);
			y = (int*)(ptr + 0x70);
		} else {
			str = *(const wchar_t**)(ptr + 0x68);
			color = (uint32_t*)(ptr + 0x74);
			length = (int*)(ptr + 0x10);
			x = (int*)(ptr + 0x18);
			y = (int*)(ptr + 0x6C);
		}
	} else if (fn == 2) {
		if (isVerMax(V_110)) {
			// remove strange flickering text on some monster names (in Jail, act1)
			str = *(const wchar_t**)(ptr + -0x14);
			color = (uint32_t*)(ptr + 0x74);
			length = (int*)(ptr + 0x14);
			x = (int*)(ptr + 0x1C);
			y = (int*)(ptr + 0x4);
			*(wchar_t*)str = 0;
		}
		else {
			str = (const wchar_t*)(ptr + 0x390);
			color = (uint32_t*)(ptr + 0x74);
			length = (int*)(ptr + 0x14);
			x = (int*)(ptr + 0x1C);
			y = (int*)(ptr + 0x4);
		}
	} else {
		str = (const wchar_t*)(ptr + 0x25C);
		color = (uint32_t*)(ptr + 0x10);
		length = (int*)(ptr - 0xC);
		x = (int*)(ptr - 0x14);
		y = (int*)(ptr + 0x8);
	}

	if (m_hovered_unit.id) {
		m_hovered_unit.id = 0;
		*length = 0;
		return;
	}

	if (str) {
		drawText(str, *x, *y, *color, 1);
		*length = 0;
	}
}
```

**Comment by bayaraa** [comment]
@rnd2k mm i see. if it fixes the issue you can create PR.
FYI maybe lenght=0 and return should be enough for skip drawing.
```cpp
		if (isVerMax(V_110)) {
			// remove strange flickering text on some monster names (in Jail, act1)
			length = (int*)(ptr + 0x14);
			*length = 0;
			return;
		}
```

**Comment by rnd2k** [comment]
Ok thank you!. I'll run another thorough test, It seems to be working. It's been this bug for quite a while (since about version 1.0.4), I apologize, I just forgot to write to you about it at the time...

**Comment by bayaraa** [comment]
Fixed in 1.3.2.
